### PR TITLE
Option to set operating mode at init_client

### PR DESCRIPTION
**Pull request by BlaiseLebreton**
## Related Issues & PRs
None

## Summary of Changes
Modified the `init_client` function that connects and configures a sensor to allow to set the operating_mode at boot and not have a forced `OPERATING_NORMAL`

## Validation
```
Running tests...
Test project /home/blebreton/svn/rtp/external/ouster-sdk/build/Release
      Start  1: osf_png_tools_test
 1/25 Test  #1: osf_png_tools_test ...............   Passed    2.49 sec
      Start  2: osf_writer_test
 2/25 Test  #2: osf_writer_test ..................   Passed    3.03 sec
      Start  3: osf_writerv2_test
 3/25 Test  #3: osf_writerv2_test ................   Passed    0.39 sec
      Start  4: osf_writer_custom_test
 4/25 Test  #4: osf_writer_custom_test ...........   Passed    0.00 sec
      Start  5: osf_file_test
 5/25 Test  #5: osf_file_test ....................   Passed    0.01 sec
      Start  6: osf_crc_test
 6/25 Test  #6: osf_crc_test .....................   Passed    0.00 sec
      Start  7: osf_file_ops_test
 7/25 Test  #7: osf_file_ops_test ................   Passed    0.00 sec
      Start  8: osf_reader_test
 8/25 Test  #8: osf_reader_test ..................   Passed    0.01 sec
      Start  9: osf_operations_test
 9/25 Test  #9: osf_operations_test ..............   Passed    0.03 sec
      Start 10: osf_basics_test
10/25 Test #10: osf_basics_test ..................   Passed    0.00 sec
      Start 11: osf_meta_streaming_info_test
11/25 Test #11: osf_meta_streaming_info_test .....   Passed    0.00 sec
      Start 12: bcompat_meta_json_test
12/25 Test #12: bcompat_meta_json_test ...........   Passed    0.07 sec
      Start 13: metadata_test
13/25 Test #13: metadata_test ....................   Passed    0.06 sec
      Start 14: lidar_scan_test
14/25 Test #14: lidar_scan_test ..................   Passed    0.07 sec
      Start 15: cartesian_test
15/25 Test #15: cartesian_test ...................   Passed    8.73 sec
      Start 16: metadata_errors_test
16/25 Test #16: metadata_errors_test .............   Passed    0.01 sec
      Start 17: pcap_test
17/25 Test #17: pcap_test ........................   Passed    0.06 sec
      Start 18: profile_extension_test
18/25 Test #18: profile_extension_test ...........   Passed    0.00 sec
      Start 19: fusa_profile_test
19/25 Test #19: fusa_profile_test ................   Passed    0.01 sec
      Start 20: parsing_benchmark_test
20/25 Test #20: parsing_benchmark_test ...........   Passed   16.69 sec
      Start 21: scan_batcher_test
21/25 Test #21: scan_batcher_test ................   Passed    0.25 sec
      Start 22: packet_writer_test
22/25 Test #22: packet_writer_test ...............   Passed    0.23 sec
      Start 23: array_view_test
23/25 Test #23: array_view_test ..................   Passed    0.56 sec
      Start 24: field_test
24/25 Test #24: field_test .......................   Passed    0.14 sec
      Start 25: point_viz_test
25/25 Test #25: point_viz_test ...................   Passed    0.00 sec

100% tests passed, 0 tests failed out of 25

Total Test time (real) =  32.85 sec
```

### CODE REVIEW

File: ouster_client/include/ouster/client.h [hunk 1]
```diff
@@ -98,6 +100,7 @@ std::shared_ptr<client> init_client(
     const std::string& hostname, const std::string& udp_dest_host,
     lidar_mode ld_mode = MODE_UNSPEC, timestamp_mode ts_mode = TIME_FROM_UNSPEC,
     int lidar_port = 0, int imu_port = 0,
+    OperatingMode operating_mode = OPERATING_NORMAL,
```
Comment (matthew-lidar) [inline ouster_client/include/ouster/client.h:103]: If you move the new argument to be after persist_config this would be a non-breaking change

**Review by matthew-lidar — COMMENTED** [review]
Looks like a good change overall! If you can move the argument and clean up the incidental whitespace changes to minimize the diff, should be good.

**Comment by BlaiseLebreton** [comment]
> Looks like a good change overall! If you can move the argument and clean up the incidental whitespace changes to minimize the diff, should be good.

Added a few commits to resolve your remarks, let me know !
All tests are passing

**Review by matthew-lidar — APPROVED** [review]

**Comment by BlaiseLebreton** [comment]
Is there anything to do on my end to solve this blocking check ?

**Review by Samahu — APPROVED** [review]

**Pull request merged**